### ISSUE TITLE
Property 'unwrap' does not exist on type 'AsyncThunkAction'

### ISSUE THREAD

**Issue by lucafaggianelli** [reduxjs/redux-toolkit#1589]
Hi

I'm trying to unwrap an AsyncThunk Action without success. I have a function that dispatches the action:

```ts
const submitZipFile = async (file: File) => {
    try {
      await dispatch(parseReports({ file })).unwrap()

      enqueueSnackbar('Reports created', { variant: 'success' })
    } catch (e) {
      enqueueSnackbar((e as Error).message, { variant: 'error' })
    }
  }
```

The `parseReports` function is created like this:

```ts
export const parseReports = createAsyncThunk<
  ProjectSource[] | null,
  { file: File },
  { state: RootState }
>('projects/parseReports', async ({ file }, { getState }) => {
  const currentProject = getState().projects.currentProject

  if (!currentProject) { return null }

  return await client.projects.parseReports(currentProject._id!, file)
})
```

But Typescript complains:
```
Property 'unwrap' does not exist on type 'AsyncThunkAction<ProjectSource[] | null, { file: File; }, { state: RootState; }>'.
```

If I remove the `unwrap()` call, the code compiles but I don't receive the response from the action or any promise rejection.

Am I doing something wrong?

**Comment by jameskuizon13** [comment]
This is what works for me: https://stackoverflow.com/a/68343963/8401517.

**Comment by MortezaNajafian** [comment]
you can use this approach and use useAppDispatch() instead of useDispatch()

export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();

**Comment by jdmaldonado** [comment]
In my case, I had to use:

```
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch();
```
It worked, but **TS** was still complaining, so I had to upgrade to the latest version.

`v1.5.0` to `v1.8.5`

I hope it works for someone.

**Comment by markerikson** [comment]
Yeah, we improved some of our TS type inference in `configureStore` as of 1.8.

**Comment by nicholascm** [comment]
@markerikson Any recommendation on how to solve this type issue if we are working in shared code that doesn't have access to the current `store` directly? (e.g. we have a shared piece of state across multiple apps, we can dispatch the actions from that shared state, but `dispatch` fails because it is the standard `useDispatch` type.

**Comment by nicholascm** [comment]
> @markerikson Any recommendation on how to solve this type issue if we are working in shared code that doesn't have access to the current `store` directly? (e.g. we have a shared piece of state across multiple apps, we can dispatch the actions from that shared state, but `dispatch` fails because it is the standard `useDispatch` type.

Actually...I just did this: 

```typescript
import { ThunkDispatch, AnyAction, Dispatch } from "@reduxjs/toolkit";
import { useDispatch } from "react-redux";

type AppDispatch = ThunkDispatch<any, undefined, AnyAction> & Dispatch<any>

export const useAppDispatch: () => AppDispatch = useDispatch;

```
